app/jobs/send_emails_after_three_weeks_job.rb

```diff
@@ -0,0 +1,20 @@
+# Runs every morning at 4am. Finds claims which are not yet checked and which
+# are three weeks old on the date specified (defaulting to today), and sends an
+# email saying we are still in the process of checking.
+
+class SendEmailsAfterThreeWeeksJob < CronJob
```

Comment:
I wondering if there might be a case where a user might potentially get two emails if this job gets requeued constantly, so it might be worth adding a boolean column to the claim that gets updated once the mail is sent?

Reply:
I feel like adding a column to track the state of this email is perhaps overkill. If we think this job is likely to fail whilst in the `each` loop, resulting in multiple mailer jobs getting queued for a given claim when this job retries then another way to mitigate against that would be to wrap the entire thing in a transaction, that way the jobs get rolled-back if this job blows up. But I'd say it would only be worth introducing that complexity if we think that there is a real risk of the code falling over part-way through queueing all the mailer jobs.

Reply:
I can't see any way the loop would fall over short of something fairly catastrophic. I think the main question is around how robust this needs to be - if it fails to run fully on one day do we _need_ to be able to pick up and send the missing emails, or do we shrug and go "they were only informational anyway"?